### ISSUE TITLE
Whatsapp and Messenger broken on WB 4

### ISSUE THREAD

**Issue by VarunAgw** [wavebox/waveboxapp#1183]
* Wavebox Version: [e.g. 3.0.0] v4 Latest
* Operating System & Version: [e.g. Windows 10, Mac OSX 10.3.4, Ubuntu 17.10] Win 10

Since today, I am getting this error on WhatsApp and Messenger.

`
Sorry, something went wrong.
We're working on getting this fixed as soon as we can.
Go Back
WhatsApp © 2018 · Help`

`
Your request couldn't be processed
Your request couldn't be processed
There was a problem with this request. We're working on getting it fixed as soon as we can.
Return home
AboutCreate adCreate PageDevelopersCareersPrivacyCookiesAdChoicesTermsHelpSettingsActivity log
Facebook © 2020
`

Note: Works fine on Chrome.

**Comment by Thomas101** [comment]
I've seen this with Messenger, but it normally indicates a sign-in problem. If you remove the account, add it again and sign back in, does it work?

Can you send a screenshot of the WhatsApp error - this is something we haven't seen before

**Comment by nlenkowski** [comment]
Just saw this today for WhatsApp as well. Deleting the account and recreating it didn't work at first, but after manually logging out of WhatsApp and logging back in (via the new account) things seemed to be working again. However, after restarting Wavebox the problem resurfaced. 

I can login to WhatsApp web via Chrome without issue. I've also tried setting a custom user agent to spoof Chrome on MacOS to no avail. Here's the error message I'm seeing.

![Wavebox  WhatsApp  Error 2020-01-08 17-40-04](https://user-images.githubusercontent.com/1290235/71997578-66aa0100-323e-11ea-8c50-c6d7b0a125eb.png)

 Thanks for looking into this!

**Comment by nlenkowski** [comment]
Just noticed that if I click "Go Back" from the error message and then click on "WhatsApp Web" from the navigation I can login just fine. A strange bug for sure. Also, forgot to mention that I'm on MacOS Catalina 10.5.2 and Wavebox 4.11.11.

**Comment by VarunAgw** [comment]
Yes. I just came here to add this. Same on messenger. Click "return home". Login on facebook. Then press the home button on address bar.

**Comment by Thomas101** [comment]
It looks like Facebook are slowly releasing an update which means some accounts are affected and some are not. The root cause in Wavebox 4 looks to be down to the way that Electron integrates with the network stack.

I've pushed out a new beta with a workaround for this, so you should be able to get it working again [More info on how to switch to the beta](https://wavebox.io/kb/beta-channel)

Also if you're interested, [Wavebox 10](https://github.com/wavebox/waveboxapp/issues/1133) is unaffected by this as it's built directly on top of Chromium

**Comment by VarunAgw** [comment]
@Thomas101 May I ask (out of curiosity) how did you solve the issue? You said it's because of the network layer but in the commit, you did something with the user agent only.

**Comment by Thomas101** [comment]
Electron uses a different implementation of some of the network layer, which means there are some oddities in parsing things like CSP headers.

The update that's rolling out to WhatsApp & Messenger seems to hit up against some of these oddities, but still change their behaviour based on the detected browser version. For the short term we've set chrome 77 as the UserAgent for these two as a workaround.

This is why WB10 isn't affected because it has a far more commonly tested and run-in network stack so we don't see spurious bugs like this and we don't have to use weird workarounds :)

Do you mind if I keep this open for now, just until it hits stable?

**Comment by VarunAgw** [comment]
Sure. I liked WB10 and jumped onto it very enthusiastically. But after a week, I felt it's not stable and ready for me yet, so I had to downgrade, unfortunately.

I will upgrade to it in future again once it's more stable.

**Comment by amig** [comment]
Had similar issue with WhatsApp in version 4.11.11. Switched to Beta (4.11.12) an problem solved.
